src/passes/enumConverter.ts

```diff
@@ -24,33 +35,43 @@ export class EnumConverter extends ASTMapper {
     return val;
   }
 
-  replacementTypeString(enumDef: EnumDefinition): string {
-    return enumToIntType(enumDef).pp();
+  visitTypeName(node: TypeName, ast: AST): void {
+    this.commonVisit(node, ast);
+    const tNode = getNodeType(node, ast.compilerVersion);
+    const replacementNode = replaceEnumType(tNode);
+    if (tNode.pp() !== replacementNode.pp()) {
+      node.typeString = generateExpressionTypeString(replacementNode);
+    }
   }
 
-  visitVariableDeclaration(node: VariableDeclaration, ast: AST): void {
-    if (node.vType instanceof UserDefinedTypeName) {
-      const enumDef = node.vType.vReferencedDeclaration;
-      if (enumDef instanceof EnumDefinition) {
-        const replacementIntType = enumToIntType(enumDef);
-        const replacementIntTypeName = typeNameFromTypeNode(replacementIntType, ast);
-        ast.replaceNode(node.vType, replacementIntTypeName);
-
-        const replacementTypeString = replacementIntType.pp();
-        node.typeString = replacementTypeString;
-      }
+  visitElementaryTypeName(node: ElementaryTypeName, ast: AST): void {
```

Comment:
I don't think visitElementaryTypeName is necessary for this pass because this pass should never affect ElementaryTypeNames. It was needed in BytesConverter because it does affect them and because the change could not be captured by that already in visitTypeName

Reply:
Makes sense. But since our `visitTypeName` function uses `getNodeType` instead of `typeNameToTypeNode`, if we come across an `ElementaryTypeName` node with an undefined typestring, it would error out.

Reply:
We should never encounter an `ElementaryTypeName` with an undefined typestring. One showing up is an error that needs dealing with at the source, not here

Reply:
Implemented a fix for above in #356